corehq/form_processor/models.py

```diff
@@ -1199,14 +1215,19 @@ def is_case_attachment(self):
 
     @property
     def is_case_rebuild(self):
-        return bool(
-            (self.TYPE_REBUILD_FORM_ARCHIVED & self.type) or
-            (self.TYPE_REBUILD_FORM_EDIT & self.type) or
-            (self.TYPE_REBUILD_USER_ARCHIVED & self.type) or
-            (self.TYPE_REBUILD_USER_REQUESTED & self.type) or
-            (self.TYPE_REBUILD_WITH_REASON & self.type) or
-            (self.TYPE_REBUILD_FORM_REPROCESS & self.type)
-        )
+        return bool(self.type & self.case_rebuild_types())
+
+    @classmethod
+    def case_rebuild_types(cls):
```

Comment:
Maybe make this `memoized` so it doesn't have to be calculated every time? (It's a constant, right?)

Reply:
sure. d44c6cd. 

thinking through, i'm not sure which would be preferred: using the memory to store the result in a dict (memoized) or doing this tiny calculation over and over again.